src/liboslexec/constfold.cpp

```diff
@@ -1062,6 +1062,44 @@ DECLFOLDER(constfold_strlen)
 
 
 
+DECLFOLDER(constfold_hash)
+{
+    // Try to turn R=strlen(s) into R=C
+    Opcode &op (rop.inst()->ops()[opnum]);
+    Symbol &S (*rop.inst()->argsymbol(op.firstarg()+1));
+    if (S.is_constant()) {
+        ASSERT (S.typespec().is_string());
+        int result = (int) (*(ustring *)S.data()).hash();
+        int cind = rop.add_constant (TypeDesc::TypeInt, &result);
```

Comment:
It's easier to

    int cind = rop.add_constant (result);

It knows result is an int. You only need the (TypeDesc,void*) version when it's not a simple type.